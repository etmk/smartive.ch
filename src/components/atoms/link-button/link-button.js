import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';

import './link-button.scss';

const mod = (modifier) => ` link-button--${modifier}`;

export const LinkButton = ({ url, text, isPrimary, isWhite, hasBorder }) => {
  const internal = /^\/(?!\/)/.test(url);
  const className = `link-button${isPrimary ? mod('primary') : ''}${isWhite ? mod('white') : ''}${
    hasBorder ? ' has-border' : ''
  }`;
  if (internal) {
    return (
      <Link to={url} className={className}>
        {text}
      </Link>
    );
  }
  return (
    <a href={url} className={className}>
      {text}
    </a>
  );
};

export default LinkButton;

LinkButton.propTypes = {
  url: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  isPrimary: PropTypes.bool,
  isWhite: PropTypes.bool,
  hasBorder: PropTypes.bool,
};

LinkButton.defaultProps = {
  isPrimary: false,
  isWhite: false,
  hasBorder: false,
};
