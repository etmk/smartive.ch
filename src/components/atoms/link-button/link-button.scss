@import "../../../scss/colors";
@import "../../../scss/responsive";

.link-button {
  text-decoration: none;
  text-align: center;
  cursor: pointer;
  background: $smartive-dark-blue;
  border: .0625rem solid $white;
  color: $white;
  display: block;
  padding: .75rem 1.5rem;
  margin: 1rem 0;
  font-weight: 300;
  transition: background .2s ease-in-out, color .2s ease-in-out, border-color .2s ease-in-out, box-shadow .2s ease-in-out;
  box-shadow: 0 .25rem 1rem 0 rgba(0, 0, 0, .2);

  &:hover {
    color: $white;
    box-shadow: 0 .75rem 2rem 0 rgba(0, 0, 0, .2);
  }

  &--primary {
    background: $smartive-blue;
    border: .0625rem solid $smartive-blue;
    box-shadow: 0 .25rem 1rem 0 rgba(0, 0, 0, .2);

    &:hover {
      background: $smartive-blue;
      border: .0625rem solid $smartive-blue;
      color: $white;
      box-shadow: 0 .75rem 2rem 0 rgba(0, 0, 0, .2);
    }
  }

  &--white {
    background: $white;
    border: .0625rem solid $white;
    color: $smartive-blue;
    box-shadow: 0 .25rem 1rem 0 rgba(0, 0, 0, .1);

    &.has-border {
      border: .0625rem solid $smartive-blue;
      box-shadow: none;

      &:hover {
        box-shadow: none;
        background: darken($white, 3%);
      }
    }

    &:hover {
      color: $smartive-blue;
      box-shadow: 0 .75rem 2rem 0 rgba(0, 0, 0, .2);
    }
  }

  @media #{$small-up} {
    display: inline-block;
  }
}

.case-teaser {
  .link-button {
    box-shadow: none;

    &:hover {
      background: #243a54;
      box-shadow: none;
    }

    &--primary {
      box-shadow: 0 .25rem 1rem 0 rgba(0, 0, 0, .2);

      &:hover {
        background: $smartive-blue;
        box-shadow: 0 .75rem 2rem 0 rgba(0, 0, 0, .4);
      }
    }
  }
}
