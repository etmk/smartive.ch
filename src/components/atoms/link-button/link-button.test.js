import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { LinkButton } from './link-button';

// This test uses the Shallow Renderer (only one JSX level deep) because the
// underlying gatsby-link relies on the router being passed in the context.
// Therefore this test doesn't test the link implementation.
const renderer = new ShallowRenderer();

describe('LinkButton', () => {
  it('renders default external link correctly', () => {
    const tree = renderer.render(<LinkButton url="http://example.com" text="LinkButton" />);
    expect(tree).toMatchSnapshot();
  });

  it('renders default internal link correctly', () => {
    const tree = renderer.render(<LinkButton url="/internal" text="LinkButton" />);
    expect(tree).toMatchSnapshot();
  });

  it('renders primary correctly', () => {
    const tree = renderer.render(<LinkButton url="http://example.com" text="Primary" isPrimary />);
    expect(tree).toMatchSnapshot();
  });

  it('renders white with border correctly', () => {
    const tree = renderer.render(<LinkButton url="http://example.com" text="White w/ Border" isWhite hasBorder />);
    expect(tree).toMatchSnapshot();
  });
});
