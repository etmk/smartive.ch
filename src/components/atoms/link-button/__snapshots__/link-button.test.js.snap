// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton renders default external link correctly 1`] = `
<a
  className="link-button"
  href="http://example.com"
>
  LinkButton
</a>
`;

exports[`LinkButton renders default internal link correctly 1`] = `
<GatsbyLink
  className="link-button"
  to="/internal"
>
  LinkButton
</GatsbyLink>
`;

exports[`LinkButton renders primary correctly 1`] = `
<a
  className="link-button link-button--primary"
  href="http://example.com"
>
  Primary
</a>
`;

exports[`LinkButton renders white with border correctly 1`] = `
<a
  className="link-button link-button--white has-border"
  href="http://example.com"
>
  White w/ Border
</a>
`;
